File: code_samples/scala/testing_unit_tests.scala
/**
 * Testing: Unit Tests
 * AI/ML Training Sample
 */

class UnitTests {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object UnitTestsExample {
  def main(args: Array[String]): Unit = {
    val instance = new UnitTests()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/testing_unit_tests.hs
-- Testing: Unit Tests
-- AI/ML Training Sample

module UnitTests where

data UnitTests = UnitTests {
    getData :: String
} deriving (Show, Eq)

process :: UnitTests -> String -> UnitTests
process obj input = obj { getData = input }

validate :: UnitTests -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = UnitTests { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
